src/main/java/randoop/generation/CoverageTracker.java

```diff
@@ -0,0 +1,200 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.jacoco.core.analysis.*;
+import org.jacoco.core.data.ExecutionDataStore;
+import org.jacoco.core.data.SessionInfoStore;
+import org.jacoco.core.instr.Instrumenter;
+import org.jacoco.core.runtime.IRuntime;
+import org.jacoco.core.runtime.LoggerRuntime;
+import org.jacoco.core.runtime.RuntimeData;
+
+/**
+ * Tracks the coverage of each method under test. Largely based on
+ * http://www.jacoco.org/jacoco/trunk/doc/examples/java/CoreTutorial.java
+ */
+public class CoverageTracker {
+  public static CoverageTracker instance = new CoverageTracker();
+
+  private final IRuntime runtime = new LoggerRuntime();
+  private final MemoryClassLoader memoryClassLoader;
+
+  private final ExecutionDataStore executionData = new ExecutionDataStore();
+  private final SessionInfoStore sessionInfos = new SessionInfoStore();
+
+  private final Instrumenter instrumenter;
+  private final RuntimeData data;
+
+  /** Map from method name to coverage details. */
+  private final Map<String, CoverageDetails> coverageDetailsMap = new HashMap<>();
+
+  /** Set of names of all the classes under test */
+  private Set<String> classesUnderTest = new HashSet<>();
+
+  /**
+   * Coverage details related to a single method under test. Tracks total number of branches and
+   * number of uncovered branches.
+   */
+  public static class CoverageDetails {
+    /** Total number of branches. */
+    public int numBranches;
+    /** Number of uncovered branches. */
+    public int uncoveredBranches;
+  }
+
+  private CoverageTracker() {
+    this.memoryClassLoader = new MemoryClassLoader(this);
+    this.instrumenter = new Instrumenter(runtime);
+    this.data = new RuntimeData();
+
+    try {
+      this.runtime.startup(data);
+    } catch (Exception e) {
+      e.printStackTrace();
+      System.exit(1);
+    }
+  }
+
+  /**
+   * Copy the set of names of classes under test.
+   *
+   * @param classesUnderTest names of all the classes under test.
+   */
+  public void setClassesUnderTest(Set<String> classesUnderTest) {
+    this.classesUnderTest = new HashSet<>(classesUnderTest);
+  }
+
+  /**
+   * Attempts to instrument a class and return the byte array representation.
+   *
+   * @param className name of the class to instrument.
+   * @return the instrumented bytes of the class, null if instrumentation failed.
+   */
+  public byte[] instrumentClass(String className) {
+    // Only instrument classes under test and their nested classes.
+    if (!classesUnderTest.contains(className)) {
+      boolean isNestedClass = false;
+
+      // Determine if this class is a nested class, in which case, the name of some
+      // existing class under test would be the prefix of this class's name.
+      for (String classUnderTest : classesUnderTest) {
+        if (className.startsWith(classUnderTest)) {
```

Comment:
This is not enough.  The next character needs to be `$` as well.
So you could test for `startsWith(classUnderTest + "$")`.